heron/api/src/java/org/apache/heron/streamlet/impl/StreamletImpl.java

```diff
@@ -205,11 +207,37 @@ public int getNumPartitions() {
     return nPartitions;
   }
 
+  /**
+   * Set the id of the stream to be used by the children nodes.
+   * Usage (assuming source is a Streamlet object with two output streams: stream1 and stream2):
+   *   source.withStream("stream1").filter(...).log();
+   *   source.withStream("stream2").filter(...).log();
+   * @param streamId The specified stream id
+   * @return Returns back the Streamlet with changed stream id
+   */
+  @SuppressWarnings("HiddenField")
+  @Override
+  public Streamlet<R> withStream(String streamId) {
```

Comment:
Also wondering the case: if `Bolt` points invalid `streamId` (not matched with `parentStreamId`) so could **validation** be required? (to cover this case either **fail-fast** or **warning message** due to redundant Bolt)?

Reply:
Agreed. Failfast is desirable for sure.

Validation is an option question for now. Currently all streamlet operators receive and emit tuples to default stream.

The stream ids are in run time information now. So we need to be careful about where the validation should be done and how to apply it.

Reply:
Heron has a check already and it runs during submission time. Not as good as compile time but should be fast enough. :)